### ISSUE TITLE
Unable to overfit on a single image (0.jpg) from Shape dataset

### ISSUE THREAD

**Issue by SubramanianKrish** [zylo117/Yet-Another-EfficientDet-Pytorch#469]
Hi! Thanks a lot for the repository. Really cool work.

background: I was able to successfully run the shape dataset training (900 images train/ 100 val) with the hp given in tutorials and get similar mAP as shown in the tutorial (0.7ish range for COCO metric)

Although this may sound weird, I am trying to train the D0 network with pre-trained weights (effectively using --head_only True) on 1 single image (0.jpg in shape dataset) as a sanity check to see if it'd overfit for that image. The train and val set both contain the same image and annotation

This is the annotation file I created from the original .json for shape data:
`{"info": {"description": "", "url": "", "version": "", "year": 2020, "contributor": "", "date_created": "2020-04-14 01:45:07.508229"}, "licenses": [{"id": 1, "name": null, "url": null}], "categories": [{"id": 1, "name": "rectangle", "supercategory": "None"}, {"id": 2, "name": "circle", "supercategory": "None"}], "images": [{"id": 0, "file_name": "0.jpg", "width": 512, "height": 512, "date_captured": "2020-04-14 01:45:07.508146", "license": 1, "coco_url": "", "flickr_url": ""}], "annotations": [{"id": 0, "image_id": 0, "category_id": 2, "iscrowd": 0, "area": 4095.9999999999986, "bbox": [200.0, 416.0, 64.0, 64.0], "segmentation": [[200.0, 416.0, 264.0, 416.0, 264.0, 480.0, 200.0, 480.0]]}, {"id": 1, "image_id": 0, "category_id": 2, "iscrowd": 0, "area": 4356.000000000001, "bbox": [424.0, 208.0, 66.0, 66.0], "segmentation": [[424.0, 208.0, 490.0, 208.0, 490.0, 274.0, 424.0, 274.0]]}]}`

The parameters for the project are exactly same: projects/shape_singleton.yml
```
project_name: shape_singleton  # also the folder name of the dataset that under data_path folder
train_set: train
val_set: val
num_gpus: 1

# mean and std in RGB order, actually this part should remain unchanged as long as your dataset is similar to coco.
mean: [0.485, 0.456, 0.406]
std: [0.229, 0.224, 0.225]

# this anchor is adapted to the dataset
anchors_scales: '[2 ** 0, 2 ** (1.0 / 3.0), 2 ** (2.0 / 3.0)]'
anchors_ratios: '[(1.0, 1.0), (1.4, 0.7), (0.7, 1.4)]'

obj_list: ['rectangle', 'circle']
```

I ran the training with the standard command from the tutorial:
`python train.py -c 0 -p shape_singleton --batch_size=1 --head_only=True --lr 1e-3 --num_epochs=50 --load_weights weights/efficientdet-d0.pth
`

Loss after 50 epochs:
`Step: 49. Epoch: 49/50. Iteration: 1/1. Cls loss: 6.29661. Reg loss: 0.02444. Total loss: 6.32105: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  2.52it/s]
Val. Epoch: 49/50. Classification loss: 19.06611. Regression loss: 0.20312. Total loss: 19.26923
`
Running eval on the trained model:
`Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.101
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.505
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.101
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.100
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.100
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.100
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.100
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = -1.000
`

As you can see, the model isn't performing so great at the COCO metric (0.5:0.95). I was expecting it to be 1 in fact. Am I doing something wrong? Or is my expectation unrealistic?

P.S: I tried to train for more epochs (200) - No luck. Doesn't improve much. Final loss was around `Step: 200. Epoch: 200/500. Iteration: 1/1. Cls loss: 0.91287. Reg loss: 0.00674. Total loss: 0.9196` and mAP@0.5IOU was at 0.126.

**Comment by zylo117** [comment]
I believe the program works, but it just takes a long time to converge with such a small dataset. 

As you can see, the loss decreses by a lot after another 150 epochs. Maybe when cls loss gets under 0.1, the result will be much better.
